internal/resources/fluentbit/resources.go

```diff
@@ -1,48 +1,290 @@
 package fluentbit
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"maps"
 	"strconv"
 	"strings"
 
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	telemetryv1alpha1 "github.com/kyma-project/telemetry-manager/apis/telemetry/v1alpha1"
+	"github.com/kyma-project/telemetry-manager/internal/configchecksum"
+	"github.com/kyma-project/telemetry-manager/internal/fluentbit/config/builder"
 	"github.com/kyma-project/telemetry-manager/internal/fluentbit/ports"
+	"github.com/kyma-project/telemetry-manager/internal/overrides"
 	commonresources "github.com/kyma-project/telemetry-manager/internal/resources/common"
+	k8sutils "github.com/kyma-project/telemetry-manager/internal/utils/k8s"
 )
 
 const (
-	checksumAnnotationKey    = "checksum/logpipeline-config"
-	istioExcludeInboundPorts = "traffic.sidecar.istio.io/excludeInboundPorts"
-	fluentbitExportSelector  = "telemetry.kyma-project.io/log-export"
-	LogAgentName             = "telemetry-fluent-bit"
+	LogAgentName              = "telemetry-fluent-bit"
+	fbSectionsConfigMapName   = LogAgentName + "-sections"
+	fbFilesConfigMapName      = LogAgentName + "-files"
+	fbLuaConfigMapName        = LogAgentName + "-luascripts"
+	fbParsersConfigMapName    = LogAgentName + "-parsers"
+	fbEnvConfigSecretName     = LogAgentName + "-env"
+	fbTLSFileConfigSecretName = LogAgentName + "-output-tls-config"
+	fbDaemonSetName           = LogAgentName
 )
 
-type DaemonSetConfig struct {
-	FluentBitImage              string
-	FluentBitConfigPrepperImage string
-	ExporterImage               string
-	PriorityClassName           string
-	MemoryLimit                 resource.Quantity
-	CPURequest                  resource.Quantity
-	MemoryRequest               resource.Quantity
+var (
+	// FluentBit
+	fbMemoryLimit   = resource.MustParse("1Gi")
+	fbCPURequest    = resource.MustParse("100m")
+	fbMemoryRequest = resource.MustParse("50Mi")
+)
+
+type Config struct {
+	PipelineDefaults builder.PipelineDefaults
+	Overrides        overrides.Config
+}
+
+type ResourceNames struct {
+	DaemonSet           types.NamespacedName
+	LuaConfigMap        types.NamespacedName
+	ParsersConfigMap    types.NamespacedName
+	SectionsConfigMap   types.NamespacedName
+	FilesConfigMap      types.NamespacedName
+	EnvConfigSecret     types.NamespacedName
+	TLSFileConfigSecret types.NamespacedName
+}
```

Comment:
Do you really need this struct? All those fields are constants, defined in the same pkg and can be accessed directly, right?